presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -30,11 +27,27 @@ private BlockUtil()
     {
     }
 
-    static void checkValidPositions(List<Integer> positions, int positionCount)
+    static void checkValidPositions(int[] positions, int offset, int length, int positionCount)
     {
-        Set<Integer> invalidPositions = positions.stream().filter(position -> position >= positionCount).collect(toSet());
-        if (!invalidPositions.isEmpty()) {
-            throw new IllegalArgumentException("Invalid positions " + invalidPositions + " in block with " + positionCount + " positions");
+        checkValidPositionsArray(positions, offset, length);
+
+        for (int i = 0; i < length; ++i) {
+            int position = positions[offset + i];
+            if (position > positionCount) {
+                throw new IllegalArgumentException("Invalid position " + position + " in block with " + positionCount + " positions");
```

Comment:
This exception message is less informative that previous one. I am fine with that. Though if you feel that it was valuable to have whole set of invalid positions in exception you can restore the logic. Performance is not important here.